modules/nw-ipfailover-configuring-check-notify-scripts.adoc

```diff
@@ -0,0 +1,103 @@
+// Module included in the following assemblies:
+//
+// * networking/configuring-ipfailover.adoc
+
+[id="nw-ipfailover-configuring-check-notify-scripts_{context}"]
+= Configuring check and notify scripts
+
+Keepalived monitors the health of the application by periodically running an optional user supplied check script. For example, the script can test a web server by issuing a request and verifying the response.
+
+////
+The script is provided through the `--check-script=<script>` option to the `oc adm ipfailover` command. The script must exit with `0` for `pass` or `1` for `fail`.
+
+By default, the check is done every two seconds, but can be changed using the `--check-interval=<seconds>` option.
+////
+
+When a check script is not provided, a simple default script is run that tests the TCP connection. This default test is suppressed when the monitor port is `0`.
+
+For each virtual IP (VIP), Keepalived keeps the state of the node. The VIP on the node may be in `master`, `backup`, or `fault` state. All VIPs on the node that are not in the `fault` state participate in the negotiation to decide which is `master` for the VIP. All of the losers enter the `backup` state. When the check script on the `master` fails, the VIP enters the `fault` state and triggers a renegotiation. When the `backup` fails, the VIP enters the `fault` state. When the check script passes again on a VIP in the `fault` state, it exits `fault` and negotiates for `master`. The resulting state is either `master` or `backup`.
+
+As cluster administrator, you can provide an optional notify script, which is called whenever the state changes. Keepalived passes the following three parameters to the script:
+
+* `$1` - `group` or `instance`
+* `$2` - Name of the `group` or `instance`
+* `$3` - The new state: `master`, `backup`, or `fault`
+
+These scripts run in the IP failover pod and use the pod file system, not the host file system. The options require the full path to the script. You must make the script available in the pod to extract the results from running the notify script. The recommended approach for providing the scripts is to use a config map.
+
+The full path names of the check and notify scripts are added to the Keepalived configuration file, `_/etc/keepalived/keepalived.conf`, which is loaded every time Keepalived starts. The scripts can be added to the pod with a config map as follows.
```

Comment:
We updated our terminology guidelines a few months ago (https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#api-object-formatting), which is why we're using "config map" and "pod" now instead of "ConfigMap" and "Pod".